### ISSUE TITLE
OpenACC and/or OpenMP support

### ISSUE THREAD

**Issue by chadmeyer** [lanl/ports-of-call#13]
Given the ubiquity of these programming models, and some downstream desires, it might be worth looking into what it would take to add these programming models as well.  

The basic programming model in both cases is `#pragma` based decoration.  I think there are now examples of at least one of these in Kokkos.

Given how this library is intended to be used, it might be easy to extend it to mark functions as needing to be compiled for device, and we might not need to worry about data locality issues.  We probably need to play with some examples in simple downstream codes in order to understand the issues more.

Relates to #12 because it opens the possibility of having Fortran decorators as well.

**Comment by Yurlungur** [comment]
Agreed---I think `OpenMP` support has to be on the list to support Pagosa in the very near future.
